diff --git a/setup.py b/setup.py
index 3df847d..2f68e6a 100644
--- a/setup.py
+++ b/setup.py
@@ -457,9 +457,15 @@ def get_extensions():
     return extensions


+# Auto-generate local version from CUDA and PyTorch versions.
+version = get_version()
+torch_version = ".".join(torch.__version__.split(".")[:2])
+if torch.version.cuda:
+    version = f"{version}+cu.{torch.version.cuda}.torch.{torch_version}"
+
 setup(
     name='mmcv' if os.getenv('MMCV_WITH_OPS', '1') == '1' else 'mmcv-lite',
-    version=get_version(),
+    version=version,
     description='OpenMMLab Computer Vision Foundation',
     keywords='computer vision',
     packages=find_packages(),
